### ISSUE TITLE
UseExpressionBodyForMethodsCodeFixProvider encountered an error…

### ISSUE THREAD

**Issue by richardcox13** [JosefPihrt/Roslynator#116]
VS 2017 Ent on C# Console Project for .NET 4.7 has just reported this (and default to disabling)

Roslynator Refactoring 2017 version 1.2.71.

The method does return a tuple:

```
private static (int red, int green, int blue) HslToRgb(double hue, double saturation, double lightness) {
    return (0, 0, 0);
}
```

Stack trace:
```
System.NullReferenceException : Object reference not set to an instance of an object.
   at Microsoft.CodeAnalysis.CSharp.Extensions.ArrowExpressionClauseSyntaxExtensions.ConvertToBlock(ArrowExpressionClauseSyntax arrowExpression,SyntaxToken semicolonToken,Boolean createReturnStatementForExpression)
   at Microsoft.CodeAnalysis.CSharp.UseExpressionBody.AbstractUseExpressionBodyCodeFixProvider`1.WithGenerateBody(TDeclaration declaration,OptionSet options)
   at Microsoft.CodeAnalysis.CSharp.UseExpressionBody.AbstractUseExpressionBodyCodeFixProvider`1.Update(TDeclaration declaration,Boolean preferExpressionBody,OptionSet options)
   at Microsoft.CodeAnalysis.CSharp.UseExpressionBody.AbstractUseExpressionBodyCodeFixProvider`1.AddEdits(SyntaxEditor editor,Diagnostic diagnostic,OptionSet options,Boolean preferExpressionBody,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.UseExpressionBody.AbstractUseExpressionBodyCodeFixProvider`1.FixAllAsync(Document document,ImmutableArray`1 diagnostics,SyntaxEditor editor,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CodeFixes.SyntaxEditorBasedCodeFixProvider.<>c__DisplayClass2_0.<FixAllAsync>b__0(SyntaxEditor editor)
   at async Microsoft.CodeAnalysis.CodeFixes.SyntaxEditorBasedCodeFixProvider.FixAllWithEditorAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputePreviewOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetPreviewOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.GetPreviewResultAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionWithNestedFlavors.<>c__DisplayClass11_0.<GetPreviewAsync>b__0(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

**Comment by JosefPihrt** [comment]
This seems to be bug in roslyn. Please report it to roslyn repo.

**Comment by jnyrup** [comment]
This looks like a duplicate of https://github.com/dotnet/roslyn/issues/17042 which was fixed in https://github.com/dotnet/roslyn/pull/17046. According to its milestone it should be fixed in VS 15.3.

As the PR was merged back in February it could be part of a 15.3 preview even though the [release notes](https://www.visualstudio.com/en-us/news/releasenotes/vs2017-preview-relnotes) does not explicitly mention this issue.

**Comment by richardcox13** [comment]
OK, thanks. Waiting on 15.3 (which surely cannot be far off now).
